tests/v2/validation/pipeline/releaseupgrade/main.go

```diff
@@ -311,3 +322,32 @@ func NewRancherClusterConfiguration(cluster pipeline.RancherCluster, newConfigNa
 
 	return
 }
+
+func NewRancherLocalClusterConfiguration(cluster pipeline.RancherCluster, newConfigName file.Name, copiedConfig []byte) (err error) {
```

Comment:
could you add a comment here for go docs? and should this function live here, since it is the main.go file, or somewhere else?

Reply:
Addressed, and added comments to both of the functions. 

Yes, this function is very specific to this CI "glue". From these `main.go` files we get binaries that are used in linked Jenkins jobs, such as HA Upgrade. The more generic and reusable part of this snippet lives here: https://github.com/rancher/rancher/blob/release/v2.8/tests/framework/extensions/pipeline/configuration.go#L17-L25